include/net/cloud.h

```diff
@@ -130,7 +145,8 @@ struct cloud_api {
 	int (*init)(const struct cloud_backend *const backend,
 		    cloud_evt_handler_t handler);
 	int (*uninit)(const struct cloud_backend *const backend);
-	int (*connect)(const struct cloud_backend *const backend);
+	int (*connect)(const struct cloud_backend *const backend,
+		       const struct cloud_lwt *const will);
```

Comment:
It would be nice to avoid an MQTT-specific pointer in the `connect` call. If the will is part of the configuration as suggested by @rlubos, that should be doable